src/types/infer.ts

```diff
@@ -878,7 +881,9 @@ export class InferType {
 
         if (innerT instanceof UserDefinedType && innerT.definition instanceof ContractDefinition) {
             const resTemplateT =
-                innerT.definition.kind === ContractKind.Interface ? typeInterface : typeContract;
+                innerT.definition.kind === ContractKind.Interface || innerT.definition.abstract
```

Comment:
It apperars that Solidity allow `intefaceId` to be used on `abstract` contracts.